builder/main.py

```
@@ -1595,6 +1595,65 @@ def download_fs_action(target, source, env):
+    for image in env.get("FLASH_EXTRA_IMAGES", []):
+        offset = image[0]
+        path = env.subst(image[1])
+        print(f" -  {str(offset).ljust(8)} | {path}")
+        cmd += [str(offset), path]
+
+    print(f" -  {app_offset.ljust(8)} | {firmware_name}")
+    cmd += [app_offset, firmware_name]
+
+    esptool = uploader_path
+    try:
+        result = subprocess.run(
```

Ruff (0.15.2)

[error] `subprocess` call: check for execution of untrusted input

(S603)